docs/prefetchfiles-nri-plugin.md

```diff
@@ -0,0 +1,156 @@
+
+## Requirements
+
+- [NRI 2.0](https://github.com/containerd/nri): Which has been integrated into containerd since [v1.7.0](https://github.com/containerd/containerd/tree/v1.7.0-beta.1).
+
+## Workflow
+
+1. Add information such as image reference and prefetch list path to annotations in pod configuration file.
+2. Run the prefetch plugin to monitoring RunPodSandbox events.
+3. The prefetch plugin fetches image reference and prefetch files path and forwards them to nydus-snapshotter.
+4. Nydus-snapshotter specifies the prefetch list when starting nydus daemon.
+5. Nydusd completes the mounting of the nydus image.
+<!-- 4. Run the prefetch plugin to fetch those information and forward them to nydus-snapshotter when the RunPodSandbox event occurs.  -->
+
+
+
+
+
+## Modify configuration file 
+
+Modify containerd's toml configuration file to enable NRI.
+```console
+sudo tee -a /etc/containerd/config.toml <<- EOF
+[plugins."io.containerd.nri.v1.nri"]
+    config_file = "/etc/nri/nri.conf"
+    disable = false
+    plugin_path = "/opt/nri/plugins"
+    socket_path = "/var/run/nri.sock"
+EOF
+```
+Containerd will load all NRI plugins in the `plugin_path` directory on startup. If you want to start an NRI plugin manually, please add the following configuration to allow other NRI plugins to connect via `socket_path`.
+
+```console
+sudo tee /etc/nri/nri.conf <<- EOF
+disableConnections: false
+EOF
+```
+
+Restart the containerd service.
+
+```console
+sudo systemctl restart containerd
+```
+Add a prefetch configuration file to specify the correct socket address. Note that this configuration file is designed for starting the NRI plugin in `pre-connection` mode. In this mode, the NRI plugin and containerd will start simultaneously.
+```console
+sudo tee /etc/nydus/prefetchConfig.toml <<- EOF
+[file_prefetch]
+socket_address = "/run/containerd-nydus/system.sock"
+EOF
+```
+Also, when manually starting the prefetch NRI plugin, the socket address can also be modified through the command line parameter `socket-addr`. Note that the priority of configuration file parameter passing is higher than that of command line parameter passing. 
+
+
+
+After start the prefetch plugin, it will monitor pod creation events. Note that NRI plugin can only be called from containerd/CRI. So creation a pod using crictl as below.
+```console
+sudo tee pod.yaml <<- EOF
+kind: pod
+metadata:
+  name: wordpress-sandbox
+  namespace: default
+  attempt: 1
+  uid: hdishd83djaidwnduwk28bcsb
+log_directory: /tmp
+annotations:
+ containerd.io/nydus-prefetch: |
+    [
+    {"image": "dockerhub.kubekey.local/dfns/wordpress:nydus_latest", "prefetch": "/path/to/wordpress_prefetch.txt"}
```

Comment:
Um. If the value is a file path, I am concerning how can we distribute the file to each K8s node?

Reply:
Maybe add an comment: specify the `prefetch` value with an URL is coming. 😉

Reply:
THANKS very much! Other comments have been modified, and I will also correct this comment in the next version commit